GVRf/Framework/jni/engine/importer/assimp_importer_jni.cpp

```diff
@@ -47,4 +86,180 @@ Java_org_gearvrf_NativeAssimpImporter_getMesh(JNIEnv * env,
             reinterpret_cast<AssimpImporter*>(jassimp_importer);
     return reinterpret_cast<jlong>(assimp_importer->getMesh(index));
 }
+
+JNIEXPORT jobject JNICALL
+Java_org_gearvrf_NativeAssimpImporter_getAssimpScene(
+        JNIEnv * env, jobject obj, jlong jassimp_importer) {
+    AssimpImporter* assimp_importer =
+            reinterpret_cast<AssimpImporter*>(jassimp_importer);
+
+    jobject jassimp_scene = NULL;
+
+    const aiScene *assimp_scene = assimp_importer->getAssimpScene();
+
+    if (!assimp_scene) {
+        LOGE("import file returned null\n");
+        goto error;
+    }
+
+    if (!AssimpImporter::create_instance(env, "org/util/jassimp/AiScene", jassimp_scene)) {
+        goto error;
+    }
+
+    if (!AssimpImporter::load_scene_graph(env, assimp_scene, jassimp_scene)) {
+        goto error;
+    }
+
+    /* jump over error handling section */
+    goto end;
+
+error:
+    {
+        jclass exception = env->FindClass("java/io/IOException");
+        if (NULL == exception) {
+            /* thats really a problem because we cannot throw in this case */
+            env->FatalError("could not throw java.io.IOException");
+        }
+        env->ThrowNew(exception, aiGetErrorString());
+    }
```

Comment:
In current GVRF, since the overall environment is triggered and controlled by VrThread which is a pthread from native environment. It means, actually the life cycle is controlled by native side but not Java side. It also implies that all exception won't be caught when you throw it. It will only make an exception hanging there until next exception sensitive JNI call is called, and then throw a native uncaught exception crash. Therefore, if you really want to stop the app and throw some useful information, please use gvr_java_stack_trace which is in util package.

What's more, imho, please use goto statement only when there is no other choice. It seems to me that it is not the case here.